src/shell/filesystem_shell.rs

```diff
@@ -177,23 +183,39 @@ impl Shell for FilesystemShell {
             },
             Some(v) => {
                 let target = v.as_string()?;
-                let path = PathBuf::from(self.path());
-                match dunce::canonicalize(path.join(target).as_path()) {
+                let path = PathBuf::from(self.path()).join(target.clone());
+
+                match dunce::canonicalize(path.as_path()) {
                     Ok(p) => p,
                     Err(_) => {
-                        return Err(ShellError::labeled_error(
+                        let err = Err(ShellError::labeled_error(
                             "Can not change to directory",
                             "directory not found",
                             v.span().clone(),
                         ));
+
+                        if path.ends_with("-") {
```

Comment:
Is it any path that ends with "-" or if the whole thing is "-"?  I'm not super familiar with the command

Reply:
Yeah, [the docs](https://doc.rust-lang.org/std/path/struct.Path.html#method.ends_with) are a bit ambiguous but it matches the whole path component:

```
error: Can not change to directory
- shell:1:15
1 |             cd foo-
  |                ^^^^ directory not found
```

Reply:
How about "cd foo/-"? Looks like that would match with ends_with